### PR TITLE
Add tsdf fusion

### PR DESCRIPTION
**Pull request by manuelli**
- adds TSDF based fusion
- switches fusion server to use tsdf based fusion by default

---
This change is [Reviewable](https://reviewable.io/reviews/robotlocomotion/spartan/280)

### CODE REVIEW

**Comment by manuelli** [comment]
jenkins please test

File: setup/docker/install_dependencies.sh [hunk 1]
```diff
@@ -33,4 +33,6 @@ pip install \
   scipy \
   pyassimp \
   pyglet \
-  plyfile
\ No newline at end of file
+  plyfile \
+  matplotlib==1.5.3 \
```
Comment (peteflorence) [inline setup/docker/install_dependencies.sh]: hm - what needed a particular matplotlib?

Comment (manuelli) [reply setup/docker/install_dependencies.sh]: This was to deal with some strange `subprocess2` related build issues with newer matplotlibs. See [this](https://github.com/matplotlib/matplotlib/issues/8361) issue

File: src/catkin_projects/fusion_server/src/fusion_server/tsdf_fusion.py [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+    dim_x = voxelGridDim[0]
+    dim_y = voxelGridDim[1]
+    dim_z = voxelGridDim[2]
+
+
+    headerSize = 8
+    tsdf_vec = np.fromfile(tsdf_bin_filename, np.float32)
+    tsdf_vec = tsdf_vec[headerSize:]
+
+    tsdf = np.reshape(tsdf_vec, voxelGridDim, order='A')
```
[the remark this answers is not in the document]
Comment (manuelli) [reply src/catkin_projects/fusion_server/src/fusion_server/tsdf_fusion.py]: fixed.

File: src/catkin_projects/fusion_server/src/fusion_server/tsdf_fusion.py [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+    print "tsdf-fusion took %d seconds" %(elapsed)
+
+
+
+def convert_tsdf_to_ply(tsdf_bin_filename, tsdf_mesh_filename):
+    """
+    Converts the tsdf binary file to a mesh file in ply format
+
+    The indexing in the tsdf is
+    (x,y,z) <--> (x + y * dim_x + z * dim_x * dim_y)
```
[the remark this answers is not in the document]
Comment (manuelli) [reply src/catkin_projects/fusion_server/src/fusion_server/tsdf_fusion.py]: yeah, a little documentation goes a long way . . .

**Comment by peteflorence** [comment]
Tested on my user on robot, works great, merging

**Pull request merged**